server/apps/publish/archive_publish.py

```diff
@@ -114,19 +115,16 @@ def update(self, id, updates, original):
                 package_id = TakesPackageService().get_take_package_id(original)
 
                 if package_id:
-                    # process the takes to form digital master file content
-                    package, package_updates = self.process_takes(updates_of_take_to_be_published=updates,
-                                                                  original_of_take_to_be_published=original,
-                                                                  package_id=package_id)
-
-                    self._set_version_last_modified_and_state(package, package_updates, last_updated)
-                    self._update_archive(package, package_updates)
-                    package.update(package_updates)
-
-                    # send it to the digital channels
-                    queued_digital = self.publish(doc=package, updates=None, target_media_type=DIGITAL)
-
-                    self.update_published_collection(published_item_id=package[config.ID_FIELD])
+                    queued_digital = self._publish_takes_package(package_id, updates, original, last_updated)
+                else:
+                    # if item is going to be sent to digital subscribers, package it as a take
+                    if self.sending_to_digital_subscribers(updates):
+                        updated = copy(original)
+                        updated.update(updates)
+                        # create a takes package
+                        package_id = TakesPackageService().package_story_as_a_take(updated, {}, None)
```

Comment:
@amagdas correct me if am wrong. Looks like we create Takes Package in both publish and correct. Shouldn't it to be 1 takes package with both published and corrected stories as package items?